src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -233,6 +234,7 @@ static void rr_subchannel_list_shutdown_and_unref(
       grpc_subchannel_notify_on_state_change(exec_ctx, sd->subchannel, NULL,
                                              NULL,
                                              &sd->connectivity_changed_closure);
+      grpc_subchannel_index_ref();
```

Comment:
The connectivity state callback already holds a ref to the LB policy.  So instead of ref'ing the subchannel index whenever we call `grpc_subchannel_notify_on_state_change()`, it would probably be simpler to just have the LB policy take a single ref to the subchannel index when it is created (in `round_robin_create()`) and then drop it when it is destroyed (in `rr_destroy()`).